NextCloud.py

```diff
@@ -155,107 +154,173 @@ def rename_group_folder(self, fid, mountpoint):
+        """ Get all shares from the user """
+        return self.requester.get(self.get_local_url())
 
     @nextcloud_method
     def get_shares_from_path(self, path, reshares=None, subfiles=None):
-        url = self.requester.get_local_url(path)
+        """
+        Get all shares from a given file/folder
 
-        if reshares is not None:
-            self.query_components.append("reshares=true")
+        Args:
+            path (str): path to file/folder
+            reshares (bool): (optional) return not only the shares from the current user but all shares from the given file
+            subfiles (bool): (optional) return all shares within a folder, given that path defines a folder
 
-        if subfiles is not None:
-            self.query_components.append("subfiles=true")
+        Returns:
 
-        return self.requester.get(url)
+        """
+        url = self.get_local_url()
+        params = {
+            "path": path,
+            "reshares": None if reshares is None else str(bool(reshares)).lower(),  # TODO: test reshares, subfiles
+            "subfiles": None if subfiles is None else str(bool(subfiles)).lower(),
+        }
+        return self.requester.get(url, params=params)
 
     @nextcloud_method
     def get_share_info(self, sid):
-        self.requester.get(self.requester.get_local_url(sid))
+        """
+        Get information about a given share
+
+        Args:
+            sid (int): share id
+
+        Returns:
+        """
+        return self.requester.get(self.get_local_url(sid))
 
     @nextcloud_method
     def create_share(
-            self, path, shareType, shareWith=None, publicUpload=None,
+            self, path, share_type, share_with=None, public_upload=None,
             password=None, permissions=None):
-        url = self.requester.get_local_url()
-        if publicUpload:
-            publicUpload = "true"
-        if (path is None or not isinstance(shareType, int)) or (shareType in [0, 1] and shareWith is None):
+        """
+        Share a file/folder with a user/group or as public link
+
+        Mandatory fields: share_type, path and share_with for share_type USER (0) or GROUP (1).
+
+        Args:
+            path (str): path to the file/folder which should be shared
+            share_type (int): ShareType attribute
+            share_with (str): user/group id with which the file should be shared
+            public_upload (bool): bool, allow public upload to a public shared folder (true/false)
+            password (str): password to protect public link Share with
+            permissions (int): sum of selected Permission attributes
+
+        Returns:
+
+        """
+        url = self.get_local_url()
+        if public_upload:
+            public_upload = "true"
+        if (path is None or not isinstance(share_type, int)) \
+                or (share_type in [ShareType.GROUP, ShareType.USER, ShareType.FEDERATED_CLOUD_SHARE]
+                    and share_with is None):
```

Comment:
I would prefer this condition to be extracted to a utility function for sake of readability (e.g. `share_parameters_dont_make_sense(path, share_type, share_with)`.